modules/openxAnalyticsAdapter.js

```diff
@@ -655,6 +664,80 @@ function pushEvent(eventType, args, auctionId) {
   }
 }
 
+function updateLoadedAdSlotsInfo(auctionId, adUnitCode, adPosition) {
+
+  if(auctionId && adUnitCode) {
+    if(!loadedAdSlots[auctionId]){
+      loadedAdSlots[auctionId] = {};
+    }
+    loadedAdSlots[auctionId][adUnitCode] = {};
+    if(adPosition) {
+      loadedAdSlots[auctionId][adUnitCode] = { adPosition: adPosition };
+    }
+  } else {
+    utils.logWarn("OX: Couldn't update loadedAdSlots information.");
+  }
+}
+
+function getLoadedAdUnitCodes(auctionId) {
+
+  return (!auctionId || !loadedAdSlots[auctionId] || typeof loadedAdSlots[auctionId] !== 'object')
+    ? [] : Object.keys(loadedAdSlots[auctionId]);
+}
+
+function pushAdPositionData(auctionId) {
+
+  if(auctionId && eventStack[auctionId] && eventStack[auctionId].events) {
+
+    let adUnitPositionMap = loadedAdSlots[auctionId];
+    if(adUnitPositionMap && JSON.stringify(adUnitPositionMap) !== "{}") {
+
+      eventStack[auctionId].events.filter(function(event) {
+        return event.eventType === auctionEndConst;
+      }).forEach(function (auctionEndEvent) {
+
+        if(auctionEndEvent.args && auctionEndEvent.args.adUnits) {
+          auctionEndEvent.args.adUnits.forEach(function (adUnitInfo) {
+            if(adUnitPositionMap[adUnitInfo.code] && adUnitPositionMap[adUnitInfo.code]["adPosition"]) {
+              adUnitInfo["adPosition"] = adUnitPositionMap[adUnitInfo.code]["adPosition"];
+            } else {
+              adUnitInfo["adPosition"] = "";
+            }
+          })
+        }
+      });
+    }
+  }
+}
+
+function getAdPositionByElementId(elementId) {
+
+  let elem = document.querySelector("#" + elementId);
+  let adPosition;
+  if (elem) {
+    let bounding = elem.getBoundingClientRect();
+    if (bounding) {
+      let windowWidth = (window.innerWidth || document.documentElement.clientWidth);
```

Comment:
Tried doing using "DOMContentLoaded" event but the callback function is not getting called every time. So finding the window size on the fly every time.